File: src/단어공부/WordStudy.java
package 단어공부;

import java.util.Scanner;

public class WordStudy {
    public static int[] getAlphabetCount(String str) {
        int[] count = new int[26];
        for (int i = 0; i < str.length(); i++) {
//            char ch = str.charAt(i);
//            if ('A' <= ch && ch <= 'Z') {
//                count[ch - 'A']++;
//            } else {
//                count[ch - 'a']++;
//            }
            count[str.charAt(i) - 'A']++;
        }

        return count;
    }

    public static int getAlphabetCount(String str, char alp) {
        int count = 0;
        for (int i = 0; i < str.length(); i++) {
            if (alp == str.charAt(i)) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.next().toUpperCase();
//        int[] count = getAlphabetCount(str);

        int maxCount = -1;
//        int maxAlphabetIndex = -1;
        char maxAlphabet = '?';
//        for (int i = 0; i < 26; i++) {
        for (char alp = 'A'; alp <= 'Z'; alp++) {
            int count = getAlphabetCount(str, alp);
            if (count > maxCount) {
                maxCount = count;
                maxAlphabet = alp;
            } else if (count == maxCount) {
                maxAlphabet = '?';
            }
//            if (count[i] > maxCount) {
//                maxCount = count[i];
////                maxAlphabetIndex = i;
//                maxAlphabet = (char) ('A' + i);
//            } else if (count[i] == maxCount) {
//                maxAlphabet = '?';
//            }
        }
//        System.out.println((char)('A' + maxAlphabetIndex));
        System.out.println(maxAlphabet);
    }
}

File: src/단어공부/WordStudy.kt
package 단어공부

import java.util.Scanner

private fun getAlphabetCount(str: String): IntArray {
    val count = IntArray(26)
    for (i in 0 until str.length) {
        val ch = str[i]
        if ('A' <= ch && ch <= 'Z') {
            count[ch - 'A']++
        } else {
            count[ch - 'a']++
        }
    }
    return count
}
fun main() {
    val sc = Scanner(System.`in`)
    val str = sc.next()

    val count = getAlphabetCount(str)

    var maxCount = -1
    var maxAlphabet = '?';

    for (i in 0 until 26) {
        if (count[i] > maxCount) {
            maxCount = count[i]
            maxAlphabet = ('A' + i).toChar()
        } else if (count[i] == maxCount) {
            maxAlphabet = '?'
        }
    }
    println(maxAlphabet)
}